### PR TITLE
mimesniff: fix many broken tests

### PR DESCRIPTION
**Pull request by KhafraDev**
Before, after
```
[mimesniff]: Completed: 1898, failed: 989, success: 909, expected failures: 989, unexpected failures: 0, skipped: 0

[mimesniff]: Completed: 1898, failed: 872, success: 1026, expected failures: 872, unexpected failures: 0, skipped: 0
```

+117 tests passing.

https://wpt.fyi/results/mimesniff/mime-types/parsing.any.html?label=master&label=experimental&product=chrome&product=firefox&product=safari&product=deno&product=node.js&aligned

### CODE REVIEW

**Comment by KhafraDev** [comment]
cc @panva the duration on wpt.fyi shows as N/A, seems like an issue with the reports?

File: lib/fetch/dataURL.js [hunk 1]
```diff
@@ -561,6 +564,33 @@ function serializeAMimeType (mimeType) {
   return serialization
 }
 
+/**
+ * @see https://fetch.spec.whatwg.org/#http-whitespace
+ * @param {string} char
+ */
+function isHTTPWhiteSpace (char) {
+  return char === '\r' || char === '\n' || char === '\t' || char === ' '
```
Comment (anonrig) [inline lib/fetch/dataURL.js:572]: Can you compare ascii numbers?

File: lib/fetch/dataURL.js [hunk 2]
```diff
@@ -561,6 +564,33 @@ function serializeAMimeType (mimeType) {
...
+/**
+ * @see https://fetch.spec.whatwg.org/#http-whitespace
+ * @param {string} str
+ */
+function removeHTTPWhitespace (str, leading = true, trailing = true) {
+  let lead = 0
+  let trail = str.length - 1
+
+  if (leading) {
+    for (; lead < str.length && isHTTPWhiteSpace(str[lead]); lead++);
```
Comment (anonrig) [inline lib/fetch/dataURL.js:584]: findIndex would be faster than this I assume?

File: lib/fetch/response.js [hunk 3]
```diff
@@ -467,7 +467,7 @@ function initializeResponse (response, init, body) {
 
   // 5. If init["headers"] exists, then fill response’s headers with init["headers"].
   if ('headers' in init && init.headers != null) {
-    fill(response[kState].headersList, init.headers)
+    fill(response[kHeaders], init.headers)
```
Comment (KhafraDev) [inline lib/fetch/response.js:470]: I think we've had some discussion about this line before, using headersList here was always incorrect, but a workaround for some bugs with data urls. However it'd also bypass header name/header value validation which *almost* sounds like a vulnerability...

File: lib/fetch/dataURL.js (hunk 2 above)
Comment (KhafraDev) [reply lib/fetch/dataURL.js:585]: Performance wasn't really my goal here, believe it or not with this monstrosity. We actually parse the WPT data set 1.5x faster than deno, and 2-4x faster than browsers lol

File: lib/fetch/dataURL.js (hunk 1 above)
Comment (KhafraDev) [reply lib/fetch/dataURL.js:573]: I don't think it's needed, or at least I'd prefer to wait for there to be benchmarks before switching over.

**Review by ronag — APPROVED** [review]

**Comment by panva** [comment]
> cc @panva the duration on wpt.fyi shows as N/A, seems like an issue with the reports?

Never bothered to figure out what triggers its inclusion.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**